core/src/main/scala/cats/data/NonEmptySeq.scala

```diff
@@ -227,11 +227,11 @@ final class NonEmptySeq[+A] private (val toSeq: Seq[A]) extends AnyVal with NonE
    * universal .toString method.
    */
   def show[AA >: A](implicit AA: Show[AA]): String =
-    s"NonEmptySeq(${toSeq.map(Show[AA].show).mkString(", ")})"
+    s"NonEmptySeq(${toSeq.iterator.map(Show[AA].show).mkString(", ")})"
```

Comment:
Oh, good change, I really like it! The only suggestion: there's `AA: Show[AA]` in the scope available already. I guess it could be a bit more efficient to call `show` on it directly rather than summon for the `Show` instance inside `map`:
```suggestion
    s"NonEmptySeq(${toSeq.iterator.map(AA.show).mkString(", ")})"
```